app/src/scripts/admin/admin.store.js

```diff
@@ -192,6 +194,66 @@ let UserStore = Reflux.createStore({
     this.update({ eventLogs, filteredLogs })
   },
 
+  // ** Filters Eventlog to smaller logs based on job status *
+  // uploading contains all jobs
+  filterLogs() {
+    let eventLogs = this.data.eventLogs
+    // let failedLogs = []
+    // let successLogs = []
+    let uploadedLogs = []
+    if (!eventLogs.length) {
+      this.getEventLogs()
+      return
+    }
+
+    for (let log of eventLogs) {
+      if (log.type != 'JOB_STARTED') {
+        let eventStatus = log.data.job.status.toLowerCase()
+
+        if (eventStatus === 'failed') {
+          // failedLogs.push(log)
+          uploadedLogs.push(log)
+        } else if (eventStatus === 'succeeded') {
+          // successLogs.push(log)
+          uploadedLogs.push(log)
+        }
+      }
+    }
+
+    this.update({ uploadedLogs }, () => {
+      this.filterYearActivity()
+    })
+  },
+
+  filterYearActivity() {
+    let uploaded = this.data.uploadedLogs
+    let yearArr = this.data.yearOptions
+    let activityLogs = this.data.activityLogs
+    let entries = {
+      FAILED: [],
+      SUCCEEDED: [],
+    }
+
+    for (let job of uploaded) {
+      let status = job.data.job.status
+      // convert date
+      let dateTime = new Date(job.date).toString()
+      let explode = dateTime.split(' ')
+      let year = explode[3]
+      if (!entries[status][year]) {
+        entries[status][year] = []
+      } else if (entries[status][year]) {
+        entries[status][year].push({ dateTime: dateTime, log: job })
+        // push to mediary array
+      }
+      if (!yearArr.includes(year)) {
+        yearArr.push(year)
+      }
+    }
+    activityLogs = entries
+    this.update({ activityLogs })
```

Comment:
This is overloading one kind of state with another. It should be its own state variable in the store.